lib/publiccloud/ssh_interactive.pm

```diff
@@ -1,67 +1,98 @@
+# Copyright 2022 SUSE LLC
 # SPDX-License-Identifier: FSFAP
 
 # Summary: Class with helpers related to SSH Interactive mode
 #
-# Maintainer: Pavel Dostal <pdostal@suse.cz>
+# Maintainer: qa-c@suse.de
 
 package publiccloud::ssh_interactive;
+use testapi qw(is_serial_terminal);
 use base Exporter;
 use testapi;
 use Utils::Backends qw(set_sshserial_dev unset_sshserial_dev);
+use utils qw(type_string_very_slow);
 use strict;
 use warnings;
+use distribution;
 
 our @EXPORT = qw(ssh_interactive_tunnel ssh_interactive_leave select_host_console);
 
+sub establish_tunnel_console {
+    # Helper call to activate a console and establish the ssh connection therein
+    my $console = $_[0];
+
+    select_console("$console");
+    # Note: Don't use script_run here! The serial terminal is set to /dev/sshserial, so every script_run will time out
+    type_string("\n~.\n", max_interval => 1);    # ensure no previous ssh connection is present
+    enter_cmd("clear");
+    enter_cmd('ssh -t sut');
+    # give the ssh connection some time to settle
+    sleep 5;
+}
+
 sub ssh_interactive_tunnel {
     # Establish the ssh interarctive tunnel to the publiccloud instance.
     # Optional arguments: 'force => 1' - reestablish tunnel, also if already established.
 
     my $instance = shift;
-    my %args = testapi::compat_args({cmd => undef}, ['cmd'], @_);
+    my %args = testapi::compat_args({force => 0, reconnect => 0}, ['force', 'reconnect'], @_);
     $args{force} //= 0;
+    $args{reconnect} //= 0;
 
     # $serialdev should be always set from os-autoinst/testapi.pm
     die '$serialdev is not set' if (!defined $serialdev || $serialdev eq '');
     return if ($args{force} != 1 && get_var('_SSH_TUNNELS_INITIALIZED', 0) == 1);
 
+    my $prev_console = current_console();
+    select_console('tunnel-console') unless ($prev_console =~ /tunnel-console/);
+
     # Prepare the environment for the SSH tunnel
     my $upload_port = get_required_var('QEMUPORT') + 1;
     my $upload_host = testapi::host_ip();
 
-    $instance->ssh_script_run(
-        # Create /dev/sshserial fifo on remote and tail|tee it to /dev/$serialdev on local
-        #   timeout => switches to script_run instead of script_output to be used so the test will not wait for the command to end
-        #   tunnel the worker port (for downloading from data/ and uploading assets / logs
-        cmd => "'rm -rf /dev/sshserial; mkfifo -m a=rwx /dev/sshserial; tail -fn +1 /dev/sshserial' 2>&1 | tee /dev/$serialdev; clear",
-        timeout => 0,
-        no_quote => 1,
-        ssh_opts => "-yt -R $upload_port:$upload_host:$upload_port",
-        username => 'root',
-    );
-    sleep 3;
-    save_screenshot;
+    # Pipe the output of the device fifo to the local serial terminal
+    background_script_run("while true; do ssh sut -R '$upload_port:$upload_host:$upload_port' 'rm -f /dev/sshserial && mkfifo -m a=rwx /dev/sshserial && tail -fn +1 /dev/sshserial' 2>&1 >/dev/$serialdev; sleep 5; done");
```

Comment:
You forgot about `-yt`
```
$ man ssh
-y  Send log information using the syslog(3) system module.  By default this information is sent to stderr.
-t  Force pseudo-terminal allocation.  This can be used to execute arbitrary screen-based programs on a remote machine, which can be very useful, e.g. when implementing menu services.
```